python/cudf/cudf/_lib/text.pyx

```diff
@@ -28,30 +30,33 @@ def read_text(object filepaths_or_buffers,
     --------
     cudf.io.text.read_text
     """
-    cdef string filename = filepaths_or_buffers.encode()
     cdef string delim = delimiter.encode()
 
     cdef unique_ptr[data_chunk_source] datasource
     cdef unique_ptr[column] c_col
+
     cdef size_t c_byte_range_offset
     cdef size_t c_byte_range_size
     cdef byte_range_info c_byte_range
 
-    if (byte_range is not None):
+    if isinstance(filepaths_or_buffers, (StringIO)):
```

Comment:
Do you want to use `TextIOBase` instead? It is a base class for text streams. `StringIO` inherits from `TextIOBase`. https://docs.python.org/3/library/io.html#io.TextIOBase
```suggestion
    if isinstance(filepaths_or_buffers, TextIOBase):
```

Pandas seems to accept a similar range of inputs for `pandas.read_table`: https://pandas.pydata.org/docs/reference/api/pandas.read_table.html

> By file-like object, we refer to objects with a `read()` method, such as a file handle (e.g. via builtin `open` function) or `StringIO`.

Any subclass of `TextIOBase` would meet those requirements, it seems.

```python
>>> isinstance(io.StringIO(), io.TextIOBase)
True
>>> isinstance(open("some_file.txt"), io.TextIOBase)
True
```

Reply:
Probably a good idea. Lets check with @randerzander.